### PR TITLE
8285773: Replace Algorithms.eatMemory(...) with WB.fullGC() in vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java

### PR DESCRIPTION
**Pull request by lmesnik**
Please review this fix. 
OOME might happen in any thread, causing test failure.

The reference tests provoke OOME only to force reference clearing.

The possible fix is to force it directly with WB.fullGC(). Another possible fix would be to throw SkippedException in DefaultUncaughtExceptionHandler. However, it might also cause OOME and still fail.

We might also want to verify that a last-ditch GC or something similar is called for each GC.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8285773](https://bugs.openjdk.java.net/browse/JDK-8285773): Replace Algorithms.eatMemory(...) with WB.fullGC() in vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to ae9f7c8b7e9289061a0351849e415890e92f89d3
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to ae9f7c8b7e9289061a0351849e415890e92f89d3

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8440/head:pull/8440` \
`$ git checkout pull/8440`

Update a local copy of the PR: \
`$ git checkout pull/8440` \
`$ git pull https://git.openjdk.java.net/jdk pull/8440/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8440`

View PR using the GUI difftool: \
`$ git pr show -t 8440`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8440.diff">https://git.openjdk.java.net/jdk/pull/8440.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java [hunk 1]
```diff
@@ -134,12 +131,12 @@ public void run() {
                     holder[i] = null;
                 }
 
-                Algorithms.eatMemory(stresser);
+                // WB.fullGC() is a lst ditch GC which should provoke
```
Comment (tschatzl) [inline test/hotspot/jtreg/vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java:134]:
```suggestion
                // WB.fullGC() is a last ditch GC which should provoke
```

**Review by tschatzl — APPROVED** [review]
Seems good.

File: test/hotspot/jtreg/vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java (hunk 1 above)
Comment (tschatzl) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java:134]: Maybe just write. `WB.fullGC()` is guaranteed to clear all kinds of weak references.

**Review by kimbarrett — APPROVED** [review]
Looks good with @tschatzl suggestions.

**Pull request closed without merging**